### PR TITLE
ScreenshotPlugin: support collection log popup

### LINKED ISSUE #13789: Collection Log Screenshot Delay
For the screenshot function, there's an option to take a screenshot when you get an item for your collection log. I think there should be an option to delay this screenshot for a second or two so it syncs up with the collection log screen pop-up for those who have that turned on.

### PR DESCRIPTION
**Pull request by superiorser9**
Closes #13789

You can have the `New addition popup` settings enabled and `New addition notification` disabled, meaning a game chat message won't be recieved by the client therefore the plugin won't trigger a screenshot.

image

This is written so if both options (`New addition popup` and `New addition notification`) are enabled, it prioritizes the popup notification and it will save a screenshot once the popup is shown, since that's what the mentioned issue suggests.

This was based of https://github.com/Nightfirecat/runelite/commits/notification-callback with a few changes as abex suggested on Discord.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 1]
```diff
@@ -622,6 +626,30 @@ else if (killType == KillType.COX_CM)
...
+	@Subscribe
+	public void onScriptPostFired(ScriptPostFired scriptPostFired)
+	{
+		if (scriptPostFired.getScriptId() != ScriptID.NOTIFICATION_START)
+		{
+			return;
+		}
+		int[] is = client.getIntStack();
+		int scriptId = is[0];
+		int arg = is[1];
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java:638]: This should also check `client.getIntStackSize()` and work from the end of the stack so we're always pulling the latest items from it. See XpDropPlugin's use of this in its ScriptPreFired hook as a reference.

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 2]
```diff
@@ -622,6 +626,30 @@ else if (killType == KillType.COX_CM)
...
+		if (scriptId != ScriptID.NOTIFICATION_DELAY || arg != 0) // !notification_delay(0)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java:639]: If the argument's value is to be checked here, it should probably be documented in `ScriptID.NOTIFICATION_DELAY`'s javadoc.

**Review by Nightfirecat — CHANGES_REQUESTED** [review]
Generally LGTM

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java (hunk 1 above)
Comment (superiorser9) [reply runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java:638]: `int[] is = client.getIntStack();`
`int intStackSize = client.getIntStackSize();`
`log.debug("IntStack {} size {}", is, intStackSize);`

It logs:
`2021-08-30 15:25:30 [Client] DEBUG n.r.c.p.screenshot.ScreenshotPlugin - IntStack [3347, 0, 43253762, 0, 43253761, 0, 512, 334, 250, -1, 52802, 19857422, 19857423, 0, 19857412, 19857411, 19857410, 7995395, 77047, 133912, 31682, 13046191, 811377, 1006, 2004581, 20269, 9718, 1595, 121254, 30753, 26613, 2000, 360979, 2208, 18920, 7995394, 145, 178, 67, 939, 433, 293, 68, 655, 10, 964, 455, 399, 400, 869, 794, 314, 440, 622, 131, 80, 0, 335, 934, 299, 896, 641, 912, 844, 671, 810, 148, 435, 17, 11, 521, 347, 65, 553, 180, 408, 482, 437, 150, 635, 382, 223, 188, 5, 351, 445, 705, 387, 175, 139, 147, 465, 823, 604, 14, 365, 30, 423, 517, 574, 192, 905, 307, 112, 416, 165, 302, 714, 607, 678, 496, 328, 402, 730, 600, 76, 159, 874, 339, 602, 60, 116, 709, 616, 723, 320, 568, 449, 26, 359, 197, 226, 111, 200, 385, 317, 161, 162, 571, 212, 968, 970, 977, 980, 992, 994, 997, 1003, 1016, 1049, 1339, 492, 1318, 1329, 695, 77, 520, 976, 597, 267, 967, 1418, 1535, 1566, 1630, 1672, 1671, 1685, 1675, 1677, 1729, 1736, 1785, 2071, 2066, 2111, 2231, 2305, 2595, 2712, 393, 2748, 2851, 2951, 3055, 3063, 182, 14286848, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] size 0`

`client.getIntStackSize()` is 0, so I can't work from the end of the stack as suggested, unless something is wrong?

**Pull request merged**